addons/beehave/nodes/leaves/blackboard_compare.gd

```diff
@@ -0,0 +1,55 @@
+## Compares two values using the specified comparison operator.
+## Returns [code]FAILURE[/code] if any of the expression fails or the
+## comparison operation returns [code]false[/code], otherwise it returns [code]SUCCESS[/code].
+@tool
+class_name BlackboardCompareCondition extends ConditionLeaf
+
+
+enum Operators {
+	EQUAL,
+	NOT_EQUAL,
+	GREATER,
+	LESS,
+	GREATER_EQUAL,
+	LESS_EQUAL,
+}
+
+
+## Expression represetning left operand.
+@export_placeholder(EXPRESSION_PLACEHOLDER) var left_operand: String = ""
+## Comparison operator.
+@export_enum("==", "!=", ">", "<", ">=", "<=") var operator: int = 0
+## Expression represetning right operand.
+@export_placeholder(EXPRESSION_PLACEHOLDER) var right_operand: String = ""
```

Comment:
I think it is important in the `BlackboardCompare` node (in cotrast to the other three) to have a bit more description on the operands. In the other nodes we have `key` and `value` fields which makes it clear what they are used for. In this node, however, they can use *any* two expressions. It would be nice if we expanded the docstring to include how they could use the value of existing keys e.g. 
*This value can be any valid GDScript expression. In order to use the existing blackboard keys for comparisson, use `get_value("key_name")` e.g. `get_value("direction").length()`*